packages/jest-runtime/src/script_transformer.js

```diff
@@ -242,12 +242,10 @@ export default class ScriptTransformer {
       }
     }
 
-    if (mapCoverage) {
-      if (!transformed.map) {
-        const inlineSourceMap = convertSourceMap.fromSource(transformed.code);
-        if (inlineSourceMap) {
-          transformed.map = inlineSourceMap.toJSON();
-        }
+    if (!transformed.map) {
+      const inlineSourceMap = convertSourceMap.fromSource(transformed.code);
```

Comment:
This is where performance issues may surface with this unguarded. `convertSourceMap.fromSource` looks for inline source maps in code files with a regex ([src](https://github.com/thlorenz/convert-source-map/blob/46a6c17c1d58f72ab0f1934669670ed179f06c72/index.js#L114)). For sufficiently huge source files, this can freeze node. It depends on the regex as well as file size, I think.

I don't have a good answer here. Some tests are probably in order; it may not be a problem. Or we may need a new sourceMaps option or something. Or we could kill inline source map support and required a separate code/map pair always.

Reply:
I hadn't looked into the source code of `convert-source-map`. I'm okay with using `mapCoverage` to guard against this. But like it was in the original PR, it should be nullifying the explicit sourcemaps emitted from preprocessors.

Also as a user of `Jest`, I would want it to come with sensible defaults, IMO defaulting map coverage to true, adding a caveat that this should be set to false in case you are facing some freeze issue is better than defaulting to false. Cuz more often than not, it will work and is incredibly important.

Reply:
Is `mapCoverage` used anymore with this changeset

Reply:
With this change, inline sourceMaps emitted from the default `babel-jest` will be read. #5257 may not be necessary.

Reply:
The point is that inline sourcempas are way slower than just using the returned sourcemap.

Reply:
Agreed. I'm actually leaning towards removing inline-source-map support. But I don't know the ramifications of it.

Reply:
Inline source map support helps when debugging tests. VS Code at least tends to behave better when everything it needs is in script it's executing.